core/android_make.go

```diff
@@ -692,6 +693,11 @@ func (g *androidMkGenerator) generateCommonActions(sb *strings.Builder, m *gener
 	sb.WriteString("\n")
 
 	cmd, args, implicits, _ := m.getArgs(ctx)
+
+	if proptools.Bool(m.Properties.Depfile) && !strings.Contains(cmd, "${depfile}") {
+		panic(errors.New(fmt.Sprintf("%s depfile is true, but ${depfile} not used in cmd", m.Name())))
```

Comment:
You can use `fmt.Errorf ` here to remove one layer of function calls.

Reply:
Also could we move it into `getArgs`, so that it's not duplicated between android_make and Linux?